### ISSUE TITLE
MessageType now limited to "error"/"message"/"open"

### ISSUE THREAD

**Issue by spyro2000** [fanout/reconnecting-eventsource#66]
The message field in a SSE channel can be freely filled by any backend. In our service, we use this field, to differentiate between different messages types like "landing-page-update" or "news-update" or "something-happened" or whatever.

![image](https://user-images.githubusercontent.com/8479925/184169880-73a5a214-e7b6-4bf3-beea-ff7b65a3566f.png)

This worked like a charm with your library (1.4.0) :)

For some reason you limited the allowed type in your updated version (I updated to 1.5.1). 

```typescript
interface EventSourceEventMap {
    "error": Event;
    "message": MessageEvent;
    "open": Event;
}
````

Now I can't subscribe to any custom events anymore if they are not named simply 'message' :(

**Comment by harmony7** [comment]
Looks like this was a bug, I'd not noticed that the event name should be any arbitrary string. I've released a 1.5.2 and this should now be possible again.

**Comment by spyro2000** [comment]
Woohoo, you rock 👍
